src/main/java/io/cdap/plugin/servicenow/sink/transform/ServiceNowTransformer.java

```diff
@@ -0,0 +1,148 @@
+import io.cdap.cdap.api.data.format.UnexpectedFormatException;
+import io.cdap.cdap.api.data.schema.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ *  Transforms structured record to JSON object
+ */
+public class ServiceNowTransformer {
+  private static final Logger LOG = LoggerFactory.getLogger(ServiceNowTransformer.class);
+  private static final int DEFAULT_SCALE = 2;
+  private static final String DATE_PATTERN = "yyyy-MM-dd";
+  private static final String DATE_TIME_PATTERN = "yyyy-MM-dd'T'HH:mm:ss";
+
+  @Nullable
+  public JsonObject transform(@Nullable StructuredRecord record) {
+    if (record == null) {
+      // Return 'null' value as it is
+      return null;
+    }
+    List<Schema.Field> fields = Objects.requireNonNull(record.getSchema().getFields(),
+                                                       "Schema fields cannot be empty");
+    JsonObject jsonObject = new JsonObject();
+    for (Schema.Field field : fields) {
+      String fieldName = field.getName();
+      Schema nonNullableSchema = field.getSchema().isNullable() ? field.getSchema().getNonNullable()
+        : field.getSchema();
+      jsonObject.addProperty(field.getName(), String.valueOf(extractValue(fieldName, record.get(field.getName()),
+                                                                          nonNullableSchema)));
+    }
+    return jsonObject;
+  }
+
+  private Object extractValue(String fieldName, Object value, Schema schema) {
+    if (value == null) {
+      // Return 'null' value as it is
+      return null;
+    }
+
+    Schema.LogicalType fieldLogicalType = schema.getLogicalType();
+    // Get values of logical types properly
+    if (fieldLogicalType != null) {
+      switch (fieldLogicalType) {
+        case TIMESTAMP_MILLIS:
+          return extractUTCDateTime(TimeUnit.MILLISECONDS.toMicros((long) value)).toString();
+        case TIMESTAMP_MICROS:
+          return extractUTCDateTime((Long) value).toString();
+        case TIME_MILLIS:
+          LocalTime time = LocalTime.ofNanoOfDay(TimeUnit.MILLISECONDS.toNanos(((Integer) value)));
+          return time.toString();
+        case TIME_MICROS:
+          LocalTime localTime = LocalTime.ofNanoOfDay(TimeUnit.MICROSECONDS.toNanos((Long) value));
+          return localTime.toString();
+        case DATE:
+          SimpleDateFormat dateFormat = new SimpleDateFormat(DATE_PATTERN);
+          Date date = new Date(Integer.parseInt(value.toString()) * TimeUnit.DAYS.toMillis(1));
+          return dateFormat.format(date);
```

Comment:
Please check if the data type mapping here in sync with the documentation (ServiceNow-batchsink.md).
For example, cdap date maps to glide_date (what is that?). Here we're mapping date to string.

Reply:
For date values, ServiceNow internally converts the string to `glide_date` (an internal type used by ServiceNow system for date fields)
Same is the case with `glide_date_time` as well.

Reply:
Just FYI... We have added handling for individual datatypes like `glide_date` and `glide_date_time` as the earlier code supported only values that were in string format.